Sources/Subs-Graphics.php

```diff
@@ -306,10 +302,69 @@ function checkGD()
 	return true;
 }
 
+function checkImagick()
+{
+	return class_exists('Imagick');
+}
+
 function resizeImageFile($source, $destination, $max_width, $max_height, $preferred_format = 0)
 {
 	global $sourcedir;
+	@ini_set('memory_limit', '90M');
+	
+	//Can we do this with ImageMagick?
+	if(checkImagick())
+	{
+		// Get the image file, we have to work with something after all
+		if(!copy($source, $destination))
+			return false;
+		try
+		{
+			$imagick = new Imagick();
+			$imagick->readImage($destination);
+			$imageformat = $imagick->getImageFormat();
+			$gif = ($imageformat === 'GIF');
+			if ($gif) // If this is a GIF, we need to coalesce all frames to ensure they're all of the same size
+				$imagick = $imagick->coalesceImages();
+			do
+			{
+				$imagick->gammaImage(0.45454545); // Work around gamma error, see http://www.ericbrasseur.org/gamma.html
+
+				// if either the width or height are null, set the current dimension to be the maximum
```

Comment:
Does it make sense to add a check at the very top of the function to just skip all processing if both `$max_width` and `$max_height` are `null`?

Reply:
No, because this function can also be used to re-encode an image that doesn't strictly need resizing (and is used in that fashion by SMF - see `reencodeImage()`)